code/modules/cm_marines/altitude_control_console.dm

```diff
@@ -100,35 +111,38 @@ GLOBAL_VAR_INIT(ship_alt, SHIP_ALT_MED)
 
 	if(.)
 		return
-	var/mob/user = usr
-	switch(action)
-		if("low_alt")
-			change_altitude(user, SHIP_ALT_LOW)
-			. = TRUE
-		if("med_alt")
-			change_altitude(user, SHIP_ALT_MED)
-			. = TRUE
-		if("high_alt")
-			change_altitude(user, SHIP_ALT_HIGH)
-			. = TRUE
-	message_admins("[key_name(user)] has changed the ship's altitude to [action].")
-
-	add_fingerprint(usr)
+	var/mob/user = ui.user
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_ALTITUDE_CHANGE))
+		message_admins("[key_name(user)] tried to change the ship's altitude, but it is still on cooldown.")
+		to_chat(user, SPAN_WARNING("Engines pending recalibration to burn again, stand by."))
+	else
+		switch(action)
+			if("low_alt")
+				change_altitude(user, SHIP_ALT_LOW)
+				. = TRUE
+			if("med_alt")
+				change_altitude(user, SHIP_ALT_MED)
+				. = TRUE
+			if("high_alt")
+				change_altitude(user, SHIP_ALT_HIGH)
+				. = TRUE
```

Comment:
```suggestion
```
Doesn't need to be selectable; there's zero benefit so you're just griefing your team doing so. And yes even if there is no front end button, that doesn't stop anyone if this string is accepted here.